components/raftstore/src/store/peer.rs

```diff
@@ -5257,9 +5273,16 @@ where
                         "status" => ?res,
                         "region_id" => self.region_id,
                         "peer_id" => self.peer.id,
-                        "progress" => ?buffer
+                        "progress" => ?buffer,
+                        "dr_auto_sync_state" => ?self.dr_auto_sync_state,
                     );
                 } else {
+                    // Once the DR replicas catch up the log during the `SyncRecover` phase, we
+                    // should enable group commit to promise `IntegrityOverLabel`. then safe
+                    // to switch to the `Sync` phase.
+                    if self.dr_auto_sync_state == DrAutoSyncState::SyncRecover {
+                        self.switch_group_commit(true, &ctx.global_replication_state)
```

Comment:
it would be called multiple times, if it's already enabled, no need to switch

Reply:
I think after `self.replication_sync = true;`, it will not be called in next time.

Reply:
oh, I see